oak_client/src/attestation.rs

```diff
@@ -109,50 +134,109 @@ impl rustls::ServerCertVerifier for RemoteAttestationVerifier {
         hostname: webpki::DNSNameRef,
         ocsp: &[u8],
     ) -> Result<rustls::ServerCertVerified, rustls::TLSError> {
-        // Only a certificate created by the Attestation Proxy is required.
-        if certs.len() != 1 {
-            return Err(rustls::TLSError::General(format!(
-                "Only a single certificate must be provided, provided {}",
-                certs.len()
-            )));
-        }
-        let certificate = &certs[0].0;
-
-        // Parse X.509 certificate encoded using DER format.
-        // https://tools.ietf.org/html/rfc7468
-        let (_, parsed_certificate) = parse_x509_certificate(&certificate).map_err(|error| {
-            rustls::TLSError::General(format!("Couldn't parse certificate: {}", error))
-        })?;
-
-        // Extract X.509 TEE extension.
-        let extensions = parsed_certificate.extensions();
-        let tee_extension = extensions.get(&TEE_EXTENSION_OID).ok_or_else(|| {
-            rustls::TLSError::General("Couldn't find X.509 TEE extension".to_string())
-        })?;
-        let tee_extension_value = parse_der(tee_extension.value)
-            .map_err(|error| {
-                rustls::TLSError::General(format!(
-                    "Couldn't parse X.509 TEE extension: {:?}",
+        for certificate in certs.iter() {
+            // Parse X.509 certificate encoded using DER format.
```

Comment:
The body of this for loop is quite long, could the functionality be encapsulated in a separate function? Then you could also do something more idiomatic using https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.try_for_each . This also avoids all the extra logging that is now done in `map_err`, you could just log in one place.

Reply:
Done